x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/30_dense_vector_script_access.yml

```diff
@@ -0,0 +1,142 @@
+---
+"Access to values of dense_vector in script":
+  - skip:
+      features: headers
+      version: " - 7.12.99"
+      reason: "Access to values of dense_vector in script was added in 7.13"
+  - do:
+      indices.create:
+        index: test-index
+        body:
+          mappings:
+            properties:
+              v:
+                type: dense_vector
+                dims: 3
+
+  - do:
+      bulk:
+        index: test-index
+        refresh: true
+        body:
+          - '{"index": {"_id": "1"}}'
+          - '{"v": [1, 1, 1]}'
+          - '{"index": {"_id": "2"}}'
+          - '{"v": [1, 1, 2]}'
+          - '{"index": {"_id": "3"}}'
+          - '{"v": [1, 1, 3]}'
+          - '{"index": {"_id": "missing_vector"}}'
+          - '{}'
+
+  # check getVectorValue() API
+  - do:
+      search:
+        body:
+          query:
+            script_score:
+              query: { "exists" : { "field" : "v" } }
+              script:
+                source: |
+                  float s = 0;
+                  for (def el : doc['v'].vectorValue) {
+                    s += el;
+                  }
+                  s;
+
+  - match: { hits.hits.0._id: "3" }
+  - match: { hits.hits.0._score: 5 }
+  - match: { hits.hits.1._id: "2" }
+  - match: { hits.hits.1._score: 4 }
+  - match: { hits.hits.2._id: "1" }
+  - match: { hits.hits.2._score: 3 }
+
+
+  # check getMagnitude() API
+  - do:
+      headers:
+        Content-Type: application/json
```

Comment:
I didn't have the chance to dig into it, but I was curious why we use this header here? I noticed it because we have a special skip for `features: headers`.

Reply:
It looks like some content types can't deal well with floating points numbers in test assertions (e.g. assertions like `gte: {hits.hits.0._score: 3.3166}`), and tests fail with errors: "java.lang.Double cannot be cast to class java.lang.Float".

I am not sure which are these content types (at test failures don't display them), but `Content-Type: application/json` always works well, and its our standard content type.

Reply:
Thanks for the explanation.